frontend/src/components/admin/ApprovedLanguagesTable.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  Tfoot,
+  Button,
+  Slider,
+  SliderTrack,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderMark,
+} from "@chakra-ui/react";
+import { ApprovedLanguagesMap, generateSortFn } from "../../utils/Utils";
+import { convertLanguageTitleCase } from "../../utils/LanguageUtils";
+
+type ApprovedLanguagesTableProps = {
+  approvedLanguagesTranslation: ApprovedLanguagesMap | undefined;
+  approvedLanguagesReview: ApprovedLanguagesMap | undefined;
+};
+
+interface ApprovedLanguageFieldSortDict {
+  [field: string]: {
+    sortFn: (a: ApprovedLanguage, b: ApprovedLanguage) => number;
+    isAscending: boolean;
+    setIsAscending: Dispatch<SetStateAction<boolean>>;
+  };
+}
+
+type ApprovedLanguage = {
+  language: string;
+  level: number;
+  role: string;
+  sliderValue: number;
+};
+
+const ApprovedLanguagesTable = ({
+  approvedLanguagesTranslation,
+  approvedLanguagesReview,
+}: ApprovedLanguagesTableProps) => {
+  const [approvedLanguages, setApprovedLanguages] = useState<
+    ApprovedLanguage[]
+  >([]);
+
+  const [isAscendingLanguage, setIsAscendingLanguage] = useState(true);
+  const [isAscendingRole, setIsAscendingRole] = useState(true);
+
+  const fieldSortDict: ApprovedLanguageFieldSortDict = {
+    language: {
+      sortFn: generateSortFn<ApprovedLanguage>("language", isAscendingLanguage),
+      isAscending: isAscendingLanguage,
+      setIsAscending: setIsAscendingLanguage,
+    },
+    role: {
+      sortFn: generateSortFn<ApprovedLanguage>("role", isAscendingRole),
+      isAscending: isAscendingRole,
+      setIsAscending: setIsAscendingRole,
+    },
+  };
+
+  const sortApprovedLanguages = (field: string) => {
+    const newApprovedLanguages = [...approvedLanguages];
+    const { sortFn, isAscending, setIsAscending } = fieldSortDict[field];
+    setIsAscending(!isAscending);
+    newApprovedLanguages.sort(sortFn);
+    setApprovedLanguages(newApprovedLanguages);
+  };
+
+  useEffect(() => {
+    if (approvedLanguagesTranslation && approvedLanguagesReview) {
+      const translationLanguages = Object.entries(
+        approvedLanguagesTranslation,
+      ).map(([language, level]) => {
+        return { language, level, role: "Translator", sliderValue: level };
+      });
+      const reviewLanguages = Object.entries(approvedLanguagesReview).map(
+        ([language, level]) => {
+          return { language, level, role: "Reviewer", sliderValue: level };
+        },
+      );
+      const combined = translationLanguages.concat(reviewLanguages);
+      sortApprovedLanguages("language");
+      setApprovedLanguages(combined);
+    }
+  }, [approvedLanguagesTranslation, approvedLanguagesReview]);
+
+  const onSliderValueChange = (index: number, newValue: number) => {
```

Comment:
nit: add a todo comment along the lines of `// TODO: call gql mutation to update new level`